component/core/src/main/java/org/exoplatform/social/core/profile/UserProfileComparator.java

```diff
@@ -46,10 +47,19 @@ private static boolean hasChanged(Profile profileToUpdate, Profile existingProfi
     Object newValue = profileToUpdate.getProperty(key);
     Object oldValue = existingProfile.getProperty(key);
 
+    //to check if the avatar is changed 
+    boolean avatarHasChanged = false;
+    if (Profile.AVATAR.equals(key)){
```

Comment:
Why should have a specific usage for the avatar field ? couldn't it be treated like other fields ?

Reply:
I can't use `hasChanged` without modifying it because the value of in existing profile avatar is null, so my solution when I have a value (<> null) for avatar in profile to update, I can detect that the value has changed.

Reply:
I didn't understand, I see that you added an extra check for the avatyer even though it is almost the same as other. could you at least add it to the return expression in case oldValue == null